ui/app/mainui/Popups.qml

```diff
@@ -206,6 +208,21 @@ QtObject {
         })
     }
 
+    function openDeleteMessagePopup(messageId, messageStore) {
+        openPopup(deleteMessageConfirmationDialogComponent,
+                  {
+                      messageId,
+                      messageStore
+                  })
+    }
+
+    function openDownloadImageDialog(imageSource) {
+        // We don't use `openPopup`, because there's no `FileDialog::closed` signal.
+        // And multiple file dialogs are (almost) ok
+        const popup = downloadImageDialogComponent.createObject(popupParent, { imageSource })
```

Comment:
This smells like a memory leak, you have to `destroy()` the object at some point

Reply:
Nice catch!
Added `destroy()` to `onAccepted` and `onRejected` slots 👌